### ISSUE TITLE
Multiple commands produce Info.plist

### ISSUE THREAD

**Issue by kidsonfilms-python-rules** [flutter/flutter#57459]
I have looked at other issues, but none solve the issue for me.
Here is my XCode Output:
```
Xcode build done.                                           12.0s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    error: Multiple commands produce
    '/Users/siddharth/Downloads/dj_torsten/build/ios/Release-iphoneos/DJTorsten.app/Info.plist':
    1) Target 'Runner' (project 'Runner') has copy command from
    '/Users/siddharth/Downloads/dj_torsten/ios/Runner/Info.plist' to
    '/Users/siddharth/Downloads/dj_torsten/build/ios/Release-iphoneos/DJTorsten.app/Info.plist'
    2) Target 'Runner' (project 'Runner') has process command with output
    '/Users/siddharth/Downloads/dj_torsten/build/ios/Release-iphoneos/DJTorsten.app/Info.plist'
    
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: The Copy Bundle Resources build phase contains this target's Info.plist file
    '/Users/siddharth/Downloads/dj_torsten/ios/Runner/Info.plist'. (in target 'Runner' from
    project 'Runner')
    warning: duplicate output file
    '/Users/siddharth/Downloads/dj_torsten/build/ios/Release-iphoneos/DJTorsten.app/Info.plist'
    on task: ProcessInfoPlistFile
    /Users/siddharth/Downloads/dj_torsten/build/ios/Release-iphoneos/DJTorsten.app/Info.plist
    /Users/siddharth/Downloads/dj_torsten/ios/Runner/Info.plist (in target 'Runner' from project
    'Runner')
```

Here is my Flutter Doctor:
```
    Doctor summary (to see all details, run flutter doctor -v):
    [✓] Flutter (Channel stable, v1.17.1, on Mac OS X 10.15.4 19E287, locale en-US)
    [✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    [✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    [✓] Android Studio (version 3.6)
    [✓] VS Code (version 1.45.1)
    [✓] Connected device (1 available)
    • No issues found!
```

I am using XCode Version 11.4.1 with MacOS 10.15.4

**Comment by ivanchaukn** [comment]
I'm having similar issues...I have a script that copies the GoogleService-Info file into Runner.app.
I tried switching to the Legacy Build System and still getting the same issue. Help is appreciated!

```
Xcode's output:
↳
    error: Multiple commands produce '/Users/ivanchau/AndroidStudioProjects/fella/build/ios/Debug-development-iphonesimulator/Runner.app/GoogleService-Info.plist':
    1) Target 'Runner' (project 'Runner') has copy command from '/Users/ivanchau/AndroidStudioProjects/fella/ios/Runner/Firebase/development/GoogleService-Info.plist' to '/Users/ivanchau/AndroidStudioProjects/fella/build/ios/Debug-development-iphonesimulator/Runner.app/GoogleService-Info.plist'
    2) Target 'Runner' (project 'Runner') has copy command from '/Users/ivanchau/AndroidStudioProjects/fella/ios/Runner/Firebase/production/GoogleService-Info.plist' to '/Users/ivanchau/AndroidStudioProjects/fella/build/ios/Debug-development-iphonesimulator/Runner.app/GoogleService-Info.plist'

    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: duplicate output file '/Users/ivanchau/AndroidStudioProjects/fella/build/ios/Debug-development-iphonesimulator/Runner.app/GoogleService-Info.plist' on task: CopyPlistFile /Users/ivanchau/AndroidStudioProjects/fella/build/ios/Debug-development-iphonesimulator/Runner.app/GoogleService-Info.plist /Users/ivanchau/AndroidStudioProjects/fella/ios/Runner/Firebase/production/GoogleService-Info.plist (in target 'Runner' from project 'Runner')
    warning: Mapping architecture armv7 to i386. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'image_picker' from project 'Pods')
    warning: Mapping architecture arm64 to x86_64. Ensure that this target's Architectures and Valid Architectures build settings are configured correctly for the iOS Simulator platform. (in target 'image_picker' from project 'Pods')
```

**Comment by jmagman** [comment]
@ivanchaukn Please never switch to the Legacy Build System, it's deprecated by Xcode and causes all kinds of issues we keep tracking down.  I'm going to hide your comment so people don't try that to work around this issue.

**Comment by jmagman** [comment]
@kidsonfilms-python-rules It looks like you incorrectly have your Info.plist in your `Copy Bundle Resources` build phase.  Did you manually edit your Xcode project?  Do you know how that got there? 
![Screen Shot 2020-05-19 at 10 36 44 AM](https://user-images.githubusercontent.com/682784/82359435-b81c8680-99bc-11ea-84ac-051807c115eb.png)

**Comment by jmagman** [comment]
@ivanchaukn I know yours looks similar, but it's a different issue since Xcode isn't processing your Info.plist (`ProcessInfoPlistFile`), it's copying both `development` and `release` flavors of `GoogleService-Info.plist` in `development`, which isn't correct.  Probably best would be to try StackOverflow or to request documentation clarification from [Firebase](https://github.com/firebase/firebase-ios-sdk/issues).

**Comment by kidsonfilms-python-rules** [comment]
Thank you for trying to figure the problem but flutter apparently created two info.plist files that are only visible on XCode

**Comment by viniSQ** [comment]
@kidsonfilms-python-rules , can you provide more details on how you've solved the problem?
I think this issue should be reopened, it's happening with several people who have upgraded flutter recently

**Comment by jmagman** [comment]
@viniSQ
You mean you literally have the same output:
```
    1) Target 'Runner' (project 'Runner') has copy command from
    '.../ios/Runner/Info.plist' to
    '.../build/ios/Release-iphoneos/....app/Info.plist'
    2) Target 'Runner' (project 'Runner') has process command with output
    '.../build/ios/Release-iphoneos/....app/Info.plist'
```
Okay to you then: It looks like you incorrectly have your Info.plist in your Copy Bundle Resources build phase. Did you manually edit your Xcode project? Do you know how that got there?

**Comment by viniSQ** [comment]
Yes, the same output. I do have Info.plist in the Copy Bundle Resources build phase, but I wasn't having the issue before upgrading the flutter version. I'll try to remove and let you know, thank you

**Comment by jmagman** [comment]
Can we get a copy of the `.metadata` file in your project?

**Comment by jmagman** [comment]
Attaching `ios/Runner.xcodeproj/project.pbxproj` would also be helpful.

**Comment by viniSQ** [comment]
Ok, I had tried to downgrade flutter. It will take some time to rebuild the tool. So I'll try your suggestion of removing Info.plist and in case it doesn't work I'll send here, thanks

**Comment by jmagman** [comment]
@viniSQ Can we get the `.metadata` and `ios/Runner.xcodeproj/project.pbxproj` file before you try?

**Comment by viniSQ** [comment]
https://drive.google.com/file/d/1I9k8p435D_XaTRDBqii5hTnmUKfG9jQX/view?usp=sharing,%20https://drive.google.com/file/d/1hSkIe6dyxTJwBirQFhAObZ7ehweZScON/view?usp=sharing

**Comment by jmagman** [comment]
@viniSQ Are you able to just attach the files to this issue?

**Comment by viniSQ** [comment]
[Archive.zip](https://github.com/flutter/flutter/files/4685674/Archive.zip)
Github didn't let me upload the files separately

**Comment by jmagman** [comment]
Your Info.plist setup looks correct to me (from just eyeballing it).  You have a `ShareSupplierImagesExtension` extension Info.plist and `Runner` Info.plist, and that looks set up correctly too.
Can I get exactly your error message?  Your build phase names are different than @kidsonfilms-python-rules.

**Comment by viniSQ** [comment]
I'll try to reproduce it, if it throws the error again I'll send it here. I didn't copy it

**Comment by kidsonfilms-python-rules** [comment]
I solved by LITERALLY finding 2 Info.plist and deleted one of them

**Comment by viniSQ** [comment]
@jmagman , the error didn't happen again. Your solution worked, thanks. I'm just still curious on why it happened after the upgrade. Was it inserted in copy bundle resources on the update of was it an issue only after the upgrade?

**Comment by kidsonfilms-python-rules** [comment]
I just tried a lot of things that I thought would help and this one worked.

**Comment by jmagman** [comment]
@kidsonfilms-python-rules Where was the Info.plist you deleted?  What was the path?  Or you deleted it from Xcode.

@viniSQ Which solution worked?  I didn't spot a second Info.plist in your Xcode project, but I may have missed it.

I'm going to reopen this (again).  I'd like to understand how this happened and prevent it for all users, not just the ones who figure out a workaround.

**Comment by kidsonfilms-python-rules** [comment]
@jmagman If you go into XCode, and open the `Runner` folder, in my case I found two `Info.plist` instead of one, so I deleted the file with less columns and that fixed it, 
It might be completely different problem in your case though.

**Comment by jmagman** [comment]
@kidsonfilms-python-rules Any idea how it got there?  The `flutter create` templates don't have two.  Are you using source control, can you tell when the second one was added to your Xcode project?

**Comment by viniSQ** [comment]
@jmagman , sorry for taking too long to answer. I appreciate that you want to solve this issue for other developers not have the same problem. 

Unfortunately, I'm not sure which solution worked, as I tried several different things. I'll try to list what I did (I may miss something)

1. I was getting the error just after upgrading flutter to 1.17 stable
2. flutter clean
3. Changed build mode to the new build system, didn't work, returned to legacy
4. It built successfully, uploaded my app to app store
5. it failed on testflight, crashing at the first moment when I opened it
6. Downgraded fluter to 1.12
7. flutter clean, didn't work
8. upgraded flutter again to 1.17
9. removed some files
flutter clean
rm ios/Podfile
rm ios/Podfile.lock
pod update
pod install
10. Remove Info.plist from Copy Bundle Resources
11. flutter build ios

After all this, it finally worked.
I think (not sure) I had tried before to remove Info.plist from Copy Bundle Resources and it didn't work.

My guess is that the solution was steps 9 and 10.

Hope it is helpful

Thank you for helping :)

**Comment by kidsonfilms-python-rules** [comment]
@jmagman I remade the Android and iOS folders ( I believe the command is `flutter create ..` just make sure you check) which then this error popped up. I don't really know what source control is so if its not in the command aforementioned above, I don't use it.

**Comment by jmagman** [comment]
I think I'm going to close this as a one-off.  If anyone has any insight into how that Info.plist got there, please file a new issue and completely fill out the template.
